lgc/patch/PatchCopyShader.cpp

```diff
@@ -340,22 +347,82 @@ void PatchCopyShader::collectGsGenericOutputInfo(Function *gsEntryPoint) {
 void PatchCopyShader::exportOutput(unsigned streamId, BuilderBase &builder) {
   auto resUsage = m_pipelineState->getShaderResourceUsage(ShaderStageCopyShader);
   auto &builtInUsage = resUsage->builtInUsage.gs;
-  const auto &genericOutByteSizes = resUsage->inOutUsage.gs.genericOutByteSizes;
-
-  // Export generic outputs
-  for (auto &byteSizeMap : genericOutByteSizes[streamId]) {
-    // <location, <component, byteSize>>
-    unsigned loc = byteSizeMap.first;
-
-    unsigned byteSize = 0;
-    for (unsigned i = 0; i < 4; ++i)
-      byteSize += byteSizeMap.second[i];
-
-    assert(byteSize % 4 == 0);
-    unsigned dwordSize = byteSize / 4;
+  auto &locInfoXfbOutInfoMap = resUsage->inOutUsage.gs.locInfoXfbOutInfoMap;
+  auto &outputLocInfoMap = resUsage->inOutUsage.outputLocInfoMap;
+
+  // Build the map between new location and output value
+  DenseMap<unsigned, Value *> newLocValueMap;
+  // Collect the output value at each mapped location in the given stream
+  const auto &newLocByteSizesMap = m_newLocByteSizesMapArray[streamId];
+  for (const auto &locByteSizePair : newLocByteSizesMap) {
+    const unsigned newLoc = locByteSizePair.first;
+    const unsigned byteSize = locByteSizePair.second;
+    assert(byteSize % 4 == 0 && byteSize <= 32);
+    const unsigned dwordSize = byteSize / 4;
     Value *outputValue =
-        loadValueFromGsVsRing(FixedVectorType::get(builder.getFloatTy(), dwordSize), loc, streamId, builder);
-    exportGenericOutput(outputValue, loc, streamId, builder);
+        loadValueFromGsVsRing(FixedVectorType::get(builder.getFloatTy(), dwordSize), newLoc, streamId, builder);
+    newLocValueMap[newLoc] = outputValue;
+  }
+
+  if (resUsage->inOutUsage.enableXfb) {
+    // Export xbf output
+    if (m_pipelineState->canPackInOut()) {
+      // With packing locations, we should collect the xfb output value at an origianl location
```

Comment:
xfb -> XFB
Change it to uppercase. There are other similar issues in this function. In comments, xfb -> XFB. Please correct them.

Reply:
Done.